lib/cache.js

```diff
@@ -0,0 +1,165 @@
+
```

Comment:
lets  add some jsdoc blocks  to the methods and class to better explain the intent